README.md

```diff
@@ -66,85 +66,118 @@ See [parse feature](features/parse.feature) for the details on using the `drafte
 
 ### C/C++ API
 
-Please refer to
-[`drafter.h`](https://github.com/apiaryio/drafter/blob/master/src/drafter.h)
-for the full API documentation. See [Drafter bindings](#bindings) for using the
-library in **other languages**.
+```c
+#include <drafter/drafter.h>
+```
 
-You can also find additional info about CAPI usage in test file named [`test/test-CAPI.c`](https://github.com/apiaryio/drafter/blob/master/test/test-CAPI.c)
+The [header](https://github.com/apiaryio/drafter/blob/master/src/drafter.h) itself is annotated with comments. [C API unit tests](https://github.com/apiaryio/drafter/blob/master/test/test-CAPI.c) provide more examples.
 
-#### Parsing a blueprint to a JSON or YAML string
+#### Parse API Blueprint into API Elements
 
-The `drafter_parse_blueprint_to` takes a source blueprint and returns the given
-blueprint parsed and serialized as [API
-Elements](http://api-elements.readthedocs.io) in YAML or JSON.
+The `drafter_parse_blueprint_to` function translates a buffered API blueprint into [API
+Elements](http://api-elements.readthedocs.io), serialized into one of its supported serialization formats.
 
 ```c
-int drafter_parse_blueprint_to(const char* source, char ** out, const drafter_options options);
+drafter_error drafter_parse_blueprint_to(
+    const char* source,
+    char** out,
+    const drafter_parse_options* parse_opts,
+    const drafter_serialize_options* serialize_opts);
+);
 ```
 
-```c
-#include <drafter/drafter.h>
+Given a pointer to a UTF-8 encoded c-string,
 
+```c
 const char* blueprint =
   "# My API\n"
   "## GET /message\n"
   "+ Response 200 (text/plain)\n"
   "\n"
   "      Hello World!\n";
+```
 
-drafter_options options;
-options.format = DRAFTER_SERIALIZE_JSON;
-options.sourcemap = true;
+##### Serialized as YAML
 
-char* result = NULL;
-if (drafter_parse_blueprint_to(blueprint, &result, options) == 0) {
-    printf("%s\n", result);
-    free(result);
+Without options, the resulting API Elements is serialized as YAML.
+
+```c
+char* yamlApie = NULL;
+if (drafter_parse_blueprint_to(blueprint, &yamlApie, NULL, NULL) == 0) {
+    printf("%s\n", yamlApie);
 }
+
+free(yamlApie);
```

Comment:
Should this go into the `if` block too? Would drafter always return an allocated block in `yamlApie` even if it failed?